pandas/io/pytables.py

```diff
@@ -3262,6 +3279,19 @@ def write_array(
         elif lib.is_np_dtype(value.dtype, "m"):
             self._handle.create_array(self.group, key, value.view("i8"))
             getattr(self.group, key)._v_attrs.value_type = "timedelta64"
+        elif isinstance(value, BaseStringArray):
+            vlarr = self._handle.create_vlarray(self.group, key, _tables().ObjectAtom())
+            vlarr.append(value.to_numpy())
+            node = getattr(self.group, key)
+            if value.dtype == StringDtype("python", np.nan):
```

Comment:
Do we need all the branches here or can you just do `str(values.dtype)`?

Reply:
Same as https://github.com/pandas-dev/pandas/pull/60663#discussion_r1906025284